test/test_dynamic_type.cpp

```diff
@@ -666,4 +770,113 @@ TEST_ASSIGN_OP(>>, >>=, RShiftAssign);
 #pragma clang diagnostic pop
 #endif
 
+namespace container_test {
+// Testing containers support by implementing the set-theoretic definition of
+// natural numbers:
+// https://en.wikipedia.org/wiki/Set-theoretic_definition_of_natural_numbers
```

Comment:
This is how mathematician define natural numbers. I think this topic is only interesting to mathematician working on set theory and logic. For other mathematician and general people, we just consider natural numbers as well defined. For our purpose, I am just implementing the following equations, because it is a good stress test if compilers works well with our `DynamicType` implementation.
![image](https://github.com/NVIDIA/Fuser/assets/1032377/62f5b0b2-55c3-4732-aa45-87e2169cf097)